airbyte-cdk/bulk/toolkits/load-iceberg-parquet/src/main/kotlin/io/airbyte/cdk/load/data/iceberg/parquet/AirbyteValueToIcebergRecord.kt

```diff
@@ -27,21 +27,25 @@ class AirbyteValueToIcebergRecord {
                     if (type.isStructType) {
                         type.asStructType().asSchema()
                     } else {
-                        throw IllegalArgumentException("ObjectValue should be mapped to ObjectType")
+                        throw IllegalArgumentException("ObjectValue should be mapped to StructType")
+                    }
+                return recordSchema
+                    .columns()
```

Comment:
Would something like this work better here?

```kotlin
    .columns()
    .mapNotNull { column ->
        airbyteValue.values[column.name()]?.let { value ->
            column.name() to convert(value, column.type())
        }
    }
    .toMap()
```

I guess I am bit concerned about the `!!` This would raise an exception, correct?

Reply:
@frifriSF59 It would, but I don't think it is possible here.  The map shows the value as non-nullable, so I don't know why Kotlin believes it could be null (I guess because the key could not exist?).  However, I do a filter first that ensures that there is an entry in the map.  Happy to change it to `mapNotNull` to be safe.

Reply:
On second pass, `mapNotNull` doesn't work here because the issue is that the lookup in the map may not be present.  The filter is protecting against this.  I think this is just a rough edge here in Kotlin's ability to detect the filter is guarding against it.  I can change it to a `getOrDefault` that returns an Airbyte `NullValue` if it is not there.  Still don't think its possible to get that default, but it will protect against it.

Reply:
All good. 🚢 it

Reply:
Updated to use `getOrDefault`.